testflinger_agent/client.py

```diff
@@ -274,3 +294,34 @@ def post_agent_data(self, data):
             self.session.post(agent_data_url, json=data, timeout=30)
         except RequestException as exc:
             logger.error(exc)
+
+    def post_influx(self, job_id, phase=None, duration=None, result=None):
+        if phase:
+            measurement = "job phase"
+            fields = {"job id": job_id, "job phase": phase}
+        if duration:
+            measurement = "phase duration"
+            fields = {"job id": job_id, "duration": duration}
+        if result:
+            measurement = "job result"
+            fields = {"job_id": job_id, "result": result}
```

Comment:
I think all of this can be combined. There's not really a concept of an overall job result, just a result (exit code) for each phase. So each time a phase is complete, we push all of this in one go - timestamp, duration, result